internal/dag/policy.go

```diff
@@ -360,3 +378,55 @@ func rateLimitPolicy(in *contour_api_v1.RateLimitPolicy) (*RateLimitPolicy, erro
 
 	return rp, nil
 }
+
+func loadBalancerHashPolicy(lbp *contour_api_v1.LoadBalancerPolicy, validCond *contour_api_v1.DetailedCondition) *LoadBalancerHashPolicy {
+	if lbp == nil {
+		return nil
+	}
+	lbhp := &LoadBalancerHashPolicy{}
+	strategy := loadBalancerPolicy(lbp)
+	switch strategy {
+	case LoadBalancerPolicyRequestHash:
+		// Do nothing, we will validate the hash policies below.
+	default:
+		lbhp.Strategy = strategy
+		return lbhp
+	}
+
+	// Map of unique header names.
+	headerHashPolicies := map[string]bool{}
+	for _, hashPolicy := range lbp.RequestHashPolicies {
+		if hashPolicy.HeaderHashOptions == nil {
+			validCond.AddWarningf("SpecError", "IgnoredField",
+				"ignoring invalid nil hash policy options")
+			continue
+		}
+		headerName := hashPolicy.HeaderHashOptions.HeaderName
+		if headerName == "" {
+			validCond.AddWarningf("SpecError", "IgnoredField",
+				"ignoring invalid header hash policy options with empty header name")
+			continue
+		}
+		if _, ok := headerHashPolicies[headerName]; ok {
+			validCond.AddWarningf("SpecError", "IgnoredField",
+				"ignoring invalid header hash policy options with duplicated header name %s", headerName)
+			continue
+		}
+
```

Comment:
we should probably use `http.CanonicalHeaderKey` and `validation.IsHTTPHeaderName` here to canonicalize and validate the header (see e.g. `headersPolicyRoute`). Also, not sure if dynamic header values is a thing in this context or not.

Reply:
good catch, added that validation 👍🏽

Reply:
we shouldn't be dealing with header values here but may be if we add regex matching/rewrite support

Reply:
oh yeah good point on header values